Editor/Editor.cpp

```diff
@@ -1864,61 +1945,190 @@ void EditorComponent::Update(float dt)
+				(HasKeyOnlySHIFT(togglewireframekey) && isShiftDown()) ||
+				(HasKeyCTRLAndSHIFT(togglewireframekey) && isCtrlDown() && isShiftDown()))
+			{
+				toggleWireframeAction();
+			}
 		}
+
 		// Enable transform tool
-		if (wi::input::Press((wi::input::BUTTON)'T'))
+		std::string toggletransformtoolkey = hotkeyssection.GetText("enable_transform_tool");
+		bool istoggletransformtoolpressed = wi::input::Press(StringToButton(toggletransformtoolkey));
+
+		auto transformToolAction = [&]()
 		{
 			translator.SetEnabled(!translator.IsEnabled());
+		};
+
+		if (istoggletransformtoolpressed)
+		{
+			if ((HasKeyOnlyCTRL(toggletransformtoolkey) && isCtrlDown()) ||
+				(HasKeyOnlySHIFT(toggletransformtoolkey) && isShiftDown()) ||
+				(HasKeyCTRLAndSHIFT(toggletransformtoolkey) && isCtrlDown() && isShiftDown()))
+			{
+				transformToolAction();
+			}
 		}
+
 		// Save
-		if (wi::input::Press((wi::input::BUTTON)'S'))
+		std::string togglesavekey = hotkeyssection.GetText("save_scene");
+		bool istogglesavepressed = wi::input::Press(StringToButton(togglesavekey));
+
+		auto saveAction = [&]()
 		{
-			if (wi::input::Down(wi::input::KEYBOARD_BUTTON_LSHIFT) || wi::input::Down(wi::input::KEYBOARD_BUTTON_RSHIFT) || GetCurrentEditorScene().path.empty())
+			if (!GetCurrentEditorScene().path.empty())
 			{
-				SaveAs();
+				Save(GetCurrentEditorScene().path);
 			}
 			else
 			{
-				Save(GetCurrentEditorScene().path);
+				SaveAs();
+			}
+		};
+
+		if (istogglesavepressed)
+		{
+			if ((HasKeyOnlyCTRL(togglesavekey) && isCtrlDown()) ||
+				(HasKeyOnlySHIFT(togglesavekey) && isShiftDown()))
+			{
+				saveAction();
+			}
+		}
+
+		// Save as
+		std::string togglesaveaskey = hotkeyssection.GetText("save_scene_as");
+		bool istogglesaveaspressed = wi::input::Press(StringToButton(togglesaveaskey));
+
+		if (istogglesaveaspressed)
+		{
+			if (HasKeyCTRLAndSHIFT(togglesaveaskey) && isCtrlDown && isShiftDown)
+			{
+				SaveAs();
 			}
 		}
+
 		// Select All
-		if (wi::input::Press((wi::input::BUTTON)'A'))
+		std::string toggleselectallkey = hotkeyssection.GetText("select_all_entitys");
+		bool istoggleselectallpressed = wi::input::Press(StringToButton(toggleselectallkey));
+
+		auto selectAllAction = [&]()
 		{
 			selectAll = true;
+		};
+
+		if (istoggleselectallpressed)
+		{
+			if ((HasKeyOnlyCTRL(toggleselectallkey) && isCtrlDown()) ||
+				(HasKeyOnlySHIFT(toggleselectallkey) && isShiftDown()) ||
+				(HasKeyCTRLAndSHIFT(toggleselectallkey) && isCtrlDown() && isShiftDown()))
+			{
+				selectAllAction();
+			}
 		}
-		// Copy/Cut
-		if (wi::input::Press((wi::input::BUTTON)'C') || wi::input::Press((wi::input::BUTTON)'X'))
+
+		// Copy
+		std::string togglecopykey = hotkeyssection.GetText("copy_action");
+		bool istogglecopypressed = wi::input::Press(StringToButton(togglecopykey));
+
+		auto copyAction = [&]()
 		{
-			auto& prevSel = translator.selectedEntitiesNonRecursive;
+			auto & prevSel = translator.selectedEntitiesNonRecursive;
```

Comment:
This kind of formatting is not what the code uses, this commit shouldn't change just formatting of existing code.